### PR TITLE
Adding Pyutilib Github Actions testing

### PR DESCRIPTION
**Pull request by mrmundt**
## Summary/Motivation:
Pyutilib can take advantage of Github Actions, hopefully so Travis/Appveyor can be retired.

## Changes proposed in this PR:
- Adding PR/Master GA Workflow

### Legal Acknowledgement

By contributing to this software project, I agree to the following terms and conditions for my contribution:

1. I agree my contributions are submitted under the BSD license.
2. I represent I am authorized to make the contributions and grant the license. If my employer has rights to intellectual property that includes these contributions, I represent that I have received permission to make contributions and grant the required license on behalf of that employer.

### CODE REVIEW

**Comment by mrmundt** [comment]
@jsiirola - The only failed GA test is because Codecov timed out. Feel free to re-trigger it (which I can't do for some reason)

File: .github/workflows/pr_master_test.yml [hunk 1]
```diff
@@ -0,0 +1,157 @@
...
+        conda config --show-sources
+        conda list --show-channel-urls
+        conda install -q -y -c conda-forge -c anaconda ${{matrix.PACKAGES}}
+        python -c 'import sys; print("::set-env name=PYTHON_EXE::%s" \
+            % (sys.executable,))'
+
+    - name: Install Pyutilib
+      run: |
+        python setup.py develop
+
```
Comment (jsiirola) [inline .github/workflows/pr_master_test.yml:137]: Can you add the following routine:
```
    - name: Set up coverage tracking
      run: |
        if test "${{matrix.TARGET}}" == win; then
            COVERAGE_BASE=${GITHUB_WORKSPACE}\\.cover
        else
            COVERAGE_BASE=${GITHUB_WORKSPACE}/.cover
        fi
        COVERAGE_RC=${COVERAGE_BASE}_rc
        echo "::set-env name=COVERAGE_RCFILE::$COVERAGE_RC"
        echo "::set-env name=COVERAGE_PROCESS_START::$COVERAGE_RC"
        cp ${GITHUB_WORKSPACE}/.coveragerc ${COVERAGE_RC}
        echo "data_file=${COVERAGE_BASE}age" >> ${COVERAGE_RC}
        SITE_PACKAGES=$($PYTHON_EXE -c "from distutils.sysconfig import \
            get_python_lib; print(get_python_lib())")
        echo "Python site-packages: $SITE_PACKAGES"
        echo 'import coverage; coverage.process_startup()' \
            > ${SITE_PACKAGES}/run_coverage_at_startup.pth

```

File: .github/workflows/pr_master_test.yml [hunk 2]
```diff
@@ -0,0 +1,157 @@
...
+  run:
+    shell: bash -l {0}
+
+env:
+  PYTHONWARNINGS: ignore::UserWarning
+  PYTHON_BASE_PKGS: >
+      coverage xlrd codecov
+
+jobs:
+  pyomo-tests:
```
Comment (jsiirola) [inline .github/workflows/pr_master_test.yml:21]: pyutilib-tests

File: .github/workflows/pr_master_test.yml [hunk 3]
```diff
@@ -0,0 +1,157 @@
...
+    #  with:
+    #    path: cache/conda
+    #    key: conda-v2-${{runner.os}}-${{matrix.python}}
+
+    - name: Pip package cache
+      uses: actions/cache@v1
+      if: matrix.PYENV == 'pip'
+      id: pip-cache
+      with:
+        path: cache/pip
```
Comment (jsiirola) [inline .github/workflows/pr_master_test.yml:69]: We also need to update `.coveragerc` to:
```
[report]
omit =
    setup.py
    */tests/*
    */tmp/*

# The [run] section must be at the end, as the build harness will add a
# "data_file" directive to the end of this file.
[run]
parallel = True
source =
    pyutilib
    examples
omit =
    # github actions creates a cache directory we don't want measured
    cache/*
```

File: .github/workflows/pr_master_test.yml [hunk 4]
```diff
@@ -0,0 +1,157 @@
...
+        python setup.py develop
+
+    - name: Run Pyutilib tests
+      run: |
+        test.pyutilib --cat=all --cov -v
+
+    - name: Process code coverage report
+      env:
+        CODECOV_NAME: ${{matrix.TARGET}}/${{matrix.python}}${{matrix.NAME}}
+      run: |
```
Comment (jsiirola) [inline .github/workflows/pr_master_test.yml:145]: Please add:
```
        coverage combine
        coverage report -i
        coverage xml -i
        i=0
        set +e
```

File: .github/workflows/pr_master_test.yml [hunk 5]
```diff
@@ -0,0 +1,157 @@
...
+        python -c 'import sys; print("::set-env name=PYTHON_EXE::%s" \
+            % (sys.executable,))'
+
+    - name: Install Pyutilib
+      run: |
+        python setup.py develop
+
+    - name: Run Pyutilib tests
+      run: |
+        test.pyutilib --cat=all --cov -v
```
Comment (jsiirola) [inline .github/workflows/pr_master_test.yml:140]: We are "hard coding" coverage, so this can just be
```
test.pyutilib -v --cat=all
```

**Pull request closed without merging**